src/main/java/io/crate/testing/CrateTestCluster.java

```diff
@@ -67,26 +67,36 @@ public class CrateTestCluster extends ExternalResource {
     private final String crateVersion;
     private final Map<String, Object> commandLineArguments;
 
+    private final int transportPortsFrom;
+    private final int transportPortsTo;
+
+    private final int httpPortsFrom;
+    private final int httpPortsTo;
+
+    private final int psqlPortsFrom;
+    private final int psqlPortsTo;
+
     private volatile CrateTestServer[] servers;
 
-    private CrateTestCluster(int numberOfNodes,
-                             String clusterName,
-                             Path workingDir,
-                             DownloadSource downloadSource,
-                             Map<String, Object> settings,
-                             String hostAddress,
-                             boolean keepWorkingDir,
-                             String crateVersion,
-                             Map<String, Object> commandLineArguments) {
-        this.numberOfNodes = numberOfNodes;
-        this.clusterName = clusterName;
-        this.workingDir = workingDir;
-        this.downloadSource = downloadSource;
-        this.settings = settings;
-        this.hostAddress = hostAddress;
-        this.keepWorkingDir = keepWorkingDir;
-        this.crateVersion = crateVersion;
-        this.commandLineArguments = commandLineArguments;
+    private CrateTestCluster(Builder builder) {
+        this.numberOfNodes = builder.numberOfNodes;
+        this.clusterName = builder.clusterName;
+        this.workingDir = builder.workingDir;
+        this.downloadSource = builder.downloadSource;
+        this.settings = builder.settings;
+        this.hostAddress = builder.hostAddress;
+        this.keepWorkingDir = builder.keepWorkingDir;
+        this.crateVersion = builder.crateVersion;
+        this.commandLineArguments = builder.commandLineArguments;
+
+        this.transportPortsFrom=builder.transportPortsFrom;
+        this.transportPortsTo=builder.transportPortsTo;
+
+        this.httpPortsFrom=builder.httpPortsFrom;
+        this.httpPortsTo=builder.httpPortsTo;
+
+        this.psqlPortsFrom=builder.httpPortsFrom;
+        this.psqlPortsTo=builder.psqlPortsTo;
     }
```

Comment:
Please don't pass in the Builder but the dedicated arguments instead, a builder patterns usually it meant to build everything but not to be passed around.

Reply:
I can if you prefer - however imho passing the builder is better and more readable than having a constructor with lots of parameters. 

Literature
- https://www.baeldung.com/java-builder-pattern 
- https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/ 
- https://www.vogella.com/tutorials/DesignPatternBuilder/article.html